pkg/api/methodHandlers.go

```diff
@@ -131,6 +138,38 @@ func (h *methodHandlers) execute(ctx context.Context, tracker attribute.Tracker,
 	return newStatus(code.Code_OK)
 }
 
+// execWrapper is responsible for dispatching calls to the manager and recovering from panics in adapter impls.
+// It also monitors the RPC's deadline for timeouts.
+func (h *methodHandlers) execWrapper(ctx context.Context, conf *aspectsupport.CombinedConfig, ab attribute.Bag) (*aspectsupport.Output, error) {
+	type result struct {
+		out *aspectsupport.Output
+		err error
+	}
+	c := make(chan result)
+	// TODO: should we use some threadpool impl instead of just spawning a thread?
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				glog.Warningf("Recovered from panic in adapter '%s'", conf.Adapter.Name)
+				if err, ok := r.(error); ok {
+					c <- result{nil, err}
+				} else {
+					c <- result{nil, fmt.Errorf("adapter '%s' panicked", conf.Adapter.Name)}
+				}
+			}
+		}()
+		o, err := h.mngr.Execute(conf, ab, h.eval)
+		c <- result{o, err}
+	}()
+
+	select {
```

Comment:
Actually, giving this more thought: if we keep the timeout check above in `execute` we can remove the check here and not spin up a new goroutine for the adapter, keeping it all in the request thread. I *think* that'd be better, thoughts?